artiq/coredevice/suservo.py

```diff
@@ -281,10 +283,92 @@ def set(self, en_out, en_iir=0, profile=0):
 
         :param en_out: RF switch enable
         :param en_iir: IIR updates enable
+        :param en_pt: Phase tracking enable
         :param profile: Active profile (0-31)
         """
         rtio_output(self.channel << 8,
-                    en_out | (en_iir << 1) | (profile << 2))
+                    en_out | (en_iir << 1) | (en_pt << 2) | (profile << 3))
+
+    @kernel
+    def set_reference_time(self, profile, fiducial_ts):
+        """Set reference time for "coherent phase mode" (see :meth:`set`).
+        This method does not advance the timeline.
+
+        Fiducial time stamp refers to the variable `T` in phase tracking mode
+        equation. See :meth:`artiq.coredevice.ad9910.AD9910.set_phase_mode`.
+        Fiducial time stamp is defined as 0 when the DDSes are updated for
+        the first time since enabling the servo.
+
+        With en_pt=1 (see :meth:`set`), the DDS output phase of this channel
+        will refer to this fiducial time stamp.
+
+        :param profile: Profile number (0-31)
+        :param fiducial_ts: Fiducial time stamp
```

Comment:
This should probably be `fiducial_mu` (or in any case, end in `_mu`) following the usual core device driver naming convention to disambiguate time in seconds and time in machine units.